src/License/Repositories/LicenseRepository.php

```diff
@@ -104,6 +110,58 @@ public function getPlatformFeePercentage(): float
         return $this->getStoredPlatformFeePercentage();
     }
 
+    /**
+     * Check if licenses are in the grace period.
+     *
+     * @unreleased
+     */
+    private function isLicenseInGracePeriod(): bool
+    {
+        $currentTime = current_time('timestamp', true);
+        
+        $lastActiveDate = get_option(LicenseOptionKeys::LAST_ACTIVE_LICENSE_DATE);
+        if ($lastActiveDate && $currentTime <= $lastActiveDate + $this->getGracePeriodInSeconds()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the date when licenses were last active.
+     * This is used to track when licenses were last valid.
```

Comment:
it sounds like this description is meant for `getLastActiveLicenseDate` - let's update the descriptions